src/com/google/enterprise/adaptor/database/TupleReader.java

```diff
@@ -151,49 +154,130 @@ protected void emitRow(ResultSet resultSet, ContentHandler handler)
       if (resultSet.getObject(col) == null) {
         atts.addAttribute("", "ISNULL", "ISNULL", "NMTOKEN", "true");
         handler.startElement("", columnName, columnName, atts);
-        // TODO: Re-examine this need for flush.
-        outWriter.flush();
         handler.endElement("", columnName, columnName);
         continue;
       }
       switch (sqlType) {
         case Types.DATE:
           Date dateValue = resultSet.getDate(col);
-          handler.startElement("", columnName, columnName, atts);
-          outWriter.write(DATEFMT.format(dateValue));
+          if (dateValue != null) {
```

Comment:
Also related to null checking, the check sqlTypeName != null 3 lines up is redundant.  It cannot be null here. (FindBugs)